agent/engine/docker_task_engine.go

```diff
@@ -2358,6 +2358,12 @@ func (engine *DockerTaskEngine) provisionContainerResourcesAwsvpc(task *apitask.
 		field.TaskID: task.GetID(),
 		"ip":         taskIP,
 	})
+	task.SetNetworkNamespace(cniConfig.ContainerNetNS)
+	// Note: By default, the interface name is set to eth0 within the CNI configs. We can also always assume that the first entry of the CNI network config to be
```

Comment:
Why is this ordering true? Or is it always config for task ENI? Can there be other Network configs returned in the slice here other than the config for task ENI?

Reply:
Is the name eth0 assigned by Agent? Do we make any assumption on the string being `eth0`? Are there any assumptions of host interface name?

Reply:
From what I understand, there can only be one ENI that's associated with a task (and it seems to be true for trunk ENI as well). For reference, this is what I'm referring to on why we can make this assumption -> https://github.com/aws/amazon-ecs-agent/blob/master/agent/api/task/task_linux.go#L285

It does seem like there are other network configs that will be appended/added but the primary task ENI/network config is what's being processed first. FWIW, we're also sort of already making an assumption that the very first entry of the task ENI will be the primary network interface here -> https://github.com/aws/amazon-ecs-agent/blob/master/agent/api/task/task.go#L2826

Reply:
>Is the name eth0 assigned by Agent? Do we make any assumption on the string being eth0? Are there any assumptions of host interface name?

Yep, the name `eth0` is the default ENI name on linux and it's set [here](https://github.com/aws/amazon-ecs-agent/blob/master/agent/ecscni/netconfig_linux.go#L130) . 

From what I can tell, seems like we might be making an assumption for `eth0`? -> https://github.com/search?q=repo%3Aaws%2Famazon-ecs-agent%20eth0&type=code

As for host mode, no there is not and we can't make this assumption as it can vary on platforms and hardware